pkg/zonemgr/zone_file.go

```diff
@@ -19,3 +25,42 @@ func NewZoneFile(fileName string) *ZoneFile {
 func (zoneFile *ZoneFile) writeFile(content string) (err error) {
 	return os.WriteFile(zoneFile.zoneFileFullName, []byte(content), zoneFilePerm)
 }
+
+func (zoneFile *ZoneFile) readFile() ([]byte, error) {
+	return os.ReadFile(zoneFile.zoneFileFullName)
+}
+
+func (zoneFile *ZoneFile) isFileExist() (*bool, error) {
+	var err error
+	isExist := false
+	if _, err = os.Stat(zoneFile.zoneFileFullName); err == nil {
+		isExist = true
+	} else if errors.Is(err, os.ErrNotExist) {
+		err = nil
+	}
+	return &isExist, err
+}
+
+func (zoneFile *ZoneFile) ReadSoaSerial() (*int, error) {
+	if isFileExist, err := zoneFile.isFileExist(); !*isFileExist || err != nil {
+		return nil, err
+	}
+	if content, err := zoneFile.readFile(); content == nil || err != nil {
```

Comment:
worth to add please the logic of isFileExist to readFile 
wdyt ?

Reply:
I considered it and preferred to separate them as 1. each function has a different responsibility 2. more readable and clear

Reply:
I understand, but on the other hand one can argue readFile should handle non existing file

Reply:
in this case it returns content=nil, just like os.ReadFile()

Reply:
if `readFile` would check before that `isFileExist` it will know to return content == nil if file doesn't exists, with err == nil, and different error in case it couldn't read the file contents

Reply:
I would leave it as it is if you don't mind, I find it more readable